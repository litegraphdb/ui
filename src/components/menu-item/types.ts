export interface MenuItemProps {
  key: string;
  icon?: React.ReactNode;
  label?: string;
  path?: string;
  children?: MenuItemProps[];
  props?: any;
}
