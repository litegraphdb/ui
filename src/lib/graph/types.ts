export interface NodeData {
  id: string;
  label: string;
  type: string;
  x: number;
  y: number;
  vx: number;
  vy: number;
  isDragging?: boolean;
}

export interface EdgeData {
  id: string;
  source: string;
  target: string;
  cost: number;
  data: string;
  sourceX: number;
  sourceY: number;
  targetX: number;
  targetY: number;
  label?: string;
}

export type HoveredElement =
  | {
      type: 'edge';
      data: EdgeData;
    }
  | {
      type: 'node';
      data: NodeData;
    }
  | null;

export interface Point {
  x: number;
  y: number;
}
