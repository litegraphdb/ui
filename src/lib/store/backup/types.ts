import { BackupMetaData } from 'litegraphdb/dist/types/types';

export type BackupType = BackupMetaData;
