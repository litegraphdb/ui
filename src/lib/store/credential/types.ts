import { CredentialMetadata } from 'litegraphdb/dist/types/types';

export type CredentialType = CredentialMetadata;
