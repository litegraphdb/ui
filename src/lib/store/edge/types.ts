import Edge from 'litegraphdb/types/models/Edge';

export type EdgeGroupWithGraph = {
  [GraphGUID: string]: EdgeType[];
};

export type EdgeType = Edge & {
  Distance?: number;
  Score?: number;
  FromName?: string;
  ToName?: string;
};
