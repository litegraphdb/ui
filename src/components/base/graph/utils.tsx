import { NodeData } from '@/lib/graph/types';
import { defaultNodeColor, nodeLightColorByType, universalNodeColors } from './constant';

const maxLegends = 16;

export const getLegendsForNodes = (nodes: NodeData[]) => {
  const priorityLegends = new Map<string, { legend: string; color: string }>();
  const customLegends = new Map<string, { legend: string; color: string }>();

  // Separate legends into priority (nodeLightColorByType) and custom
  nodes.forEach((node) => {
    if (Object.keys(nodeLightColorByType).includes(node.type)) {
      priorityLegends.set(node.type, { legend: node.type, color: nodeLightColorByType[node.type] });
    } else {
      // Only add if not already present to avoid changing colors for same legend
      if (!customLegends.has(node.type)) {
        customLegends.set(node.type, {
          legend: node.type,
          color: universalNodeColors[customLegends.size % universalNodeColors.length],
        });
      }
    }
  });

  const priorityLegendsSize = priorityLegends.size;
  // If there are nodes with unknown types, add "Unknown" to priority legends
  const isMoreThanMaxLegends = customLegends.size + priorityLegendsSize > maxLegends;

  // Convert to arrays for easier manipulation
  const priorityArray = Array.from(priorityLegends.values());
  const customArray = Array.from(customLegends.values()).slice(
    0,
    maxLegends - (priorityArray.length + (isMoreThanMaxLegends ? 1 : 0))
  );

  // Combine priority legends first, then custom legends
  const legends = [
    ...priorityArray,
    ...customArray,
    ...(isMoreThanMaxLegends ? [{ legend: 'Unknown', color: defaultNodeColor }] : []),
  ];
  const mapLegends: Record<string, { legend: string; color: string }> = {};
  legends.forEach((legend) => {
    mapLegends[legend.legend] = legend;
  });
  return mapLegends;
};
