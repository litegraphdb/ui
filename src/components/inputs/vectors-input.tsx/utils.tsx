export const convertVectorsToAPIRecord = (
  vectors?: Array<{ Model: string; Dimensionality: number; Content: string; Vectors: number[] }>
) => {
  return (
    vectors?.map((vector: any) => ({
      ...vector,
      Dimensionality: Number(vector.Dimensionality),
      Vectors: vector.Vectors || [],
    })) || []
  );
};
