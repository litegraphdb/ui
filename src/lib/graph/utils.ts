import type { NodeData, Point } from './types';

export function isPointInNode(point: Point, node: NodeData, radius: number = 10): boolean {
  const dx = point.x - node.x;
  const dy = point.y - node.y;
  return dx * dx + dy * dy <= radius * radius;
}

export function getNodeAtPosition(nodes: NodeData[], point: Point): NodeData | null {
  return nodes.find((node) => isPointInNode(point, node)) || null;
}

export function transformToOptions<T extends { GUID: string; name?: string; Name?: string }>(
  data?: T[] | null,
  labelField: keyof T = 'name' // Field to use for label in options
) {
  return data
    ? data?.map((item) => ({
        value: item.GUID,
        label: (item[labelField] as string) || item.Name || item.GUID,
      }))
    : [];
}
