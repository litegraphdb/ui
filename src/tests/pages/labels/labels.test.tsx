import '@testing-library/jest-dom';
import React from 'react';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import LabelPage from '@/app/dashboard/[tenantId]/labels/page';
import { createMockInitialState } from '../../store/mockStore';
import { mockLabelData, mockGraphGUID } from '../mockData';
import { handlers } from './handler';
import { setupServer } from 'msw/node';
import { commonHandlers } from '@/tests/handler';
import { renderWithRedux } from '@/tests/store/utils';
import AddEditLabel from '@/page/labels/components/AddEditLabel';
import DeleteLabel from '@/page/labels/components/DeleteLabel';
import { LabelMetadataForTable } from '@/page/labels/types';

// Mock react-hot-toast
jest.mock('react-hot-toast', () => ({
  success: jest.fn(),
  error: jest.fn(),
}));

const server = setupServer(...handlers, ...commonHandlers);

describe('LabelsPage', () => {
  beforeAll(() => server.listen());
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => server.close());

  it('renders the labels page', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<LabelPage />, initialState, undefined, true);

    await waitFor(() => {
      expect(screen.getByText(/labels/i)).toBeVisible();
      expect(screen.getByRole('button', { name: /create label/i })).toBeVisible();
    });
    expect(container).toMatchSnapshot('initial table state');
  });

  it('should display Create Label button', () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<LabelPage />, initialState, undefined, true);

    const createButton = screen.getByRole('button', { name: /create label/i });
    expect(createButton).toBeVisible();
    expect(createButton).toMatchSnapshot();
  });

  it('should handle LabelMetadataForTable type correctly', () => {
    const labelMetadata: LabelMetadataForTable = {
      GUID: 'label-123',
      TenantGUID: 'tenant-456',
      GraphGUID: 'graph-789',
      NodeGUID: 'node-101',
      EdgeGUID: 'edge-123',
      Label: 'priority',
      key: 'priority',
      CreatedUtc: '2024-01-01T00:00:00Z',
      LastUpdateUtc: '2024-01-01T12:00:00Z',
      NodeName: 'Test Node',
      EdgeName: 'Test Edge',
    };

    expect(labelMetadata.GUID).toBe('label-123');
    expect(labelMetadata.NodeName).toBe('Test Node');
    expect(labelMetadata.EdgeName).toBe('Test Edge');
    expect(labelMetadata.key).toBe('priority');
  });

  it('should create a label and should be visible in the table', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(
      <AddEditLabel
        isAddEditLabelVisible={true}
        setIsAddEditLabelVisible={() => {}}
        label={null}
        selectedGraph={mockGraphGUID}
      />,
      initialState,
      undefined,
      true
    );

    const modal = await screen.findByTestId('add-edit-label-modal');
    expect(modal).toBeInTheDocument();

    const labelInput = screen.getByPlaceholderText(/enter label label/i);

    fireEvent.change(labelInput, { target: { value: mockLabelData[0].Label } });

    await waitFor(() => {
      expect(labelInput.value).toBe('Test Label');
    });

    await waitFor(
      () => {
        const createButton = screen.getByRole('button', { name: /ok/i });
        expect(createButton).not.toBeDisabled();
      },
      { timeout: 5000 }
    );

    const createButton = screen.getByRole('button', { name: /ok/i });
    fireEvent.click(createButton);

    expect(container).toMatchSnapshot('after label creation form submission');
  }, 8000);

  it('should update label successfully', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(
      <AddEditLabel
        isAddEditLabelVisible={true}
        setIsAddEditLabelVisible={() => {}}
        label={mockLabelData[0]}
        selectedGraph={mockGraphGUID}
      />,
      initialState,
      undefined,
      true
    );

    const modal = await screen.findByTestId('add-edit-label-modal');
    expect(modal).toBeInTheDocument();

    expect(screen.getByText('Edit Label')).toBeInTheDocument();
    const labelInput = screen.getByPlaceholderText(/enter label label/i);
    expect(labelInput.value).toBe(mockLabelData[0].Label);
    const updatedLabelValue = 'Updated Test Label';
    fireEvent.change(labelInput, { target: { value: updatedLabelValue } });

    await waitFor(() => {
      expect(labelInput.value).toBe(updatedLabelValue);
    });

    await waitFor(
      () => {
        const submitButton = screen.getByRole('button', { name: /ok/i });
        expect(submitButton).not.toBeDisabled();
      },
      { timeout: 5000 }
    );

    const submitButton = screen.getByRole('button', { name: /ok/i });
    fireEvent.click(submitButton);

    await waitFor(() => {
      // Check that the toast.success was called instead of looking for the text in DOM
      const { success } = require('react-hot-toast');
      expect(success).toHaveBeenCalledWith('Label updated successfully');
    });

    expect(container).toMatchSnapshot('final state after label update');
  }, 8000);

  it('should delete label successfully', async () => {
    const initialState = createMockInitialState();
    const mockSetIsDeleteModelVisible = jest.fn();
    const mockSetSelectedLabel = jest.fn();
    const mockOnLabelDeleted = jest.fn();

    const { container } = renderWithRedux(
      <DeleteLabel
        title={`Are you sure you want to delete "${mockLabelData[0].Label}" label?`}
        paragraphText="This action will delete label."
        isDeleteModelVisible={true}
        setIsDeleteModelVisible={mockSetIsDeleteModelVisible}
        selectedLabel={mockLabelData[0]}
        setSelectedLabel={mockSetSelectedLabel}
        onLabelDeleted={mockOnLabelDeleted}
      />,
      initialState,
      undefined,
      true
    );

    const modal = await screen.findByTestId('delete-label-modal');
    expect(modal).toBeInTheDocument();

    expect(
      screen.getByText('Are you sure you want to delete "Test Label" label?')
    ).toBeInTheDocument();

    expect(screen.getByText('This action will delete label.')).toBeInTheDocument();

    const deleteButton = screen.getByRole('button', { name: /delete/i });
    expect(deleteButton).toBeInTheDocument();
    expect(deleteButton).toHaveClass('ant-btn-dangerous'); // Ant Design danger button class

    const cancelButton = screen.getByRole('button', { name: /cancel/i });
    expect(cancelButton).toBeInTheDocument();

    fireEvent.click(deleteButton);

    await waitFor(() => {
      // Check that the toast.success was called instead of looking for the text in DOM
      const { success } = require('react-hot-toast');
      expect(success).toHaveBeenCalledWith('Label deleted successfully');
    });

    expect(mockSetIsDeleteModelVisible).toHaveBeenCalledWith(false);
    expect(mockSetSelectedLabel).toHaveBeenCalledWith(null);
    expect(mockOnLabelDeleted).toHaveBeenCalled();

    expect(container).toMatchSnapshot('after label deletion');
  }, 8000);
});
