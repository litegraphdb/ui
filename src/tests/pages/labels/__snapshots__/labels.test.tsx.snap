// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelsPage renders the labels page: initial table state 1`] = `
<div>
  <main
    class="ant-layout-content pageContainer css-dev-only-do-not-override-98ntnt"
    id="labels"
  >
    <div
      class="pageTitleContainer ant-flex css-dev-only-do-not-override-98ntnt ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-space-between ant-flex-gap-small"
    >
      <span
        class="ant-typography css-dev-only-do-not-override-98ntnt"
        data-testid="heading"
        style="font-weight: 600; font-size: 16px;"
      >
        Labels
      </span>
      <button
        class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-link"
        style="font-weight: 600;"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="plus-square"
            class="anticon anticon-plus-square"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus-square"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
              />
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
          </span>
        </span>
        <span>
          Create Label
        </span>
      </button>
    </div>
    <div
      class="pageContent"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-98ntnt"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-98ntnt"
        >
          <div>
            <div
              aria-busy="true"
              aria-live="polite"
              class="ant-spin ant-spin-spinning css-dev-only-do-not-override-98ntnt"
            >
              <span
                class="ant-spin-dot-holder"
              >
                <span
                  class="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                </span>
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-table ant-table-empty css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 250px;"
                      />
                      <col
                        style="width: 350px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          aria-label="Label"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Label
                                </span>
                                <span
                                  class="ant-table-column-sorter ant-table-column-sorter-full"
                                >
                                  <span
                                    aria-hidden="true"
                                    class="ant-table-column-sorter-inner"
                                  >
                                    <span
                                      aria-label="caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      aria-label="caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              GUID
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="Node"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Node
                                </span>
                                <span
                                  class="ant-table-column-sorter ant-table-column-sorter-full"
                                >
                                  <span
                                    aria-hidden="true"
                                    class="ant-table-column-sorter-inner"
                                  >
                                    <span
                                      aria-label="caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      aria-label="caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="Edge"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Edge
                                </span>
                                <span
                                  class="ant-table-column-sorter ant-table-column-sorter-full"
                                >
                                  <span
                                    aria-hidden="true"
                                    class="ant-table-column-sorter-inner"
                                  >
                                    <span
                                      aria-label="caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      aria-label="caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="Created UTC"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Created UTC
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                aria-hidden="true"
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Actions
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="6"
                        >
                          <div
                            class="css-dev-only-do-not-override-98ntnt ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <title>
                                  Simple Empty
                                </title>
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    cx="32"
                                    cy="33"
                                    fill="#f5f5f5"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    fill-rule="nonzero"
                                    stroke="#d9d9d9"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                      fill="#fafafa"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              class="ant-empty-description"
                            >
                              No data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;

exports[`LabelsPage should create a label and should be visible in the table: after label creation form submission 1`] = `
<div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;

exports[`LabelsPage should delete label successfully: after label deletion 1`] = `
<div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  >
    <div
      class="go4109123758"
      style="left: 0px; right: 0px; display: flex; position: absolute; transition: all 230ms cubic-bezier(.21,1.02,.73,1); transform: translateY(0px); top: 0px; justify-content: center;"
    >
      <div
        class="go2072408551"
        style="opacity: 0;"
      >
        <div
          class="go685806154"
        >
          <div
            class="go1858758034"
          />
          <div
            class="go1579819456"
          >
            <div
              class="go2344853693"
            />
          </div>
        </div>
        <div
          aria-live="polite"
          class="go3958317564"
          role="status"
        >
          Label deleted successfully
        </div>
      </div>
    </div>
    <div
      class="go4109123758"
      style="left: 0px; right: 0px; display: flex; position: absolute; transition: all 230ms cubic-bezier(.21,1.02,.73,1); transform: translateY(0px); top: 0px; justify-content: center;"
    >
      <div
        class="go2072408551"
        style="opacity: 0;"
      >
        <div
          class="go685806154"
        >
          <div
            class="go1858758034"
          />
          <div
            class="go1579819456"
          >
            <div
              class="go2344853693"
            />
          </div>
        </div>
        <div
          aria-live="polite"
          class="go3958317564"
          role="status"
        >
          Label updated successfully
        </div>
      </div>
    </div>
    <div
      class="go4109123758"
      style="left: 0px; right: 0px; display: flex; position: absolute; transition: all 230ms cubic-bezier(.21,1.02,.73,1); transform: translateY(0px); top: 0px; justify-content: center;"
    >
      <div
        class="go2072408551"
        style="opacity: 0;"
      >
        <div
          class="go685806154"
        >
          <div
            class="go1858758034"
          />
          <div
            class="go1579819456"
          >
            <div
              class="go2344853693"
            />
          </div>
        </div>
        <div
          aria-live="polite"
          class="go3958317564"
          role="status"
        >
          Label created successfully
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LabelsPage should display Create Label button 1`] = `
<button
  class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-link"
  style="font-weight: 600;"
  type="button"
>
  <span
    class="ant-btn-icon"
  >
    <span
      aria-label="plus-square"
      class="anticon anticon-plus-square"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus-square"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
        />
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
        />
      </svg>
    </span>
  </span>
  <span>
    Create Label
  </span>
</button>
`;

exports[`LabelsPage should update label successfully: final state after label update 1`] = `
<div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  >
    <div
      class="go4109123758"
      style="left: 0px; right: 0px; display: flex; position: absolute; transition: all 230ms cubic-bezier(.21,1.02,.73,1); transform: translateY(0px); top: 0px; justify-content: center;"
    >
      <div
        class="go2072408551"
        style="opacity: 0;"
      >
        <div
          class="go685806154"
        >
          <div
            class="go1858758034"
          />
          <div
            class="go1579819456"
          >
            <div
              class="go2344853693"
            />
          </div>
        </div>
        <div
          aria-live="polite"
          class="go3958317564"
          role="status"
        >
          Label updated successfully
        </div>
      </div>
    </div>
    <div
      class="go4109123758"
      style="left: 0px; right: 0px; display: flex; position: absolute; transition: all 230ms cubic-bezier(.21,1.02,.73,1); transform: translateY(0px); top: 0px; justify-content: center;"
    >
      <div
        class="go2072408551"
        style="opacity: 0;"
      >
        <div
          class="go685806154"
        >
          <div
            class="go1858758034"
          />
          <div
            class="go1579819456"
          >
            <div
              class="go2344853693"
            />
          </div>
        </div>
        <div
          aria-live="polite"
          class="go3958317564"
          role="status"
        >
          Label created successfully
        </div>
      </div>
    </div>
  </div>
</div>
`;
